client/app/services/auto-refresh.service.js

```diff
@@ -0,0 +1,44 @@
+/** @ngInject */
+export function AutoRefreshFactory() {
+  var callbacks = [];
+  var service = {
+    triggerAutoRefresh: triggerAutoRefresh,
+    listenForAutoRefresh: listenForAutoRefresh,
+    callbacks: callbacks,
+  };
+
+  function triggerAutoRefresh(data) {
+    angular.forEach(callbacks, (function(callback) {
```

Comment:
consider using  arrow function and `.forEach` markup

would result in the following rewrite:
```
callbacks.forEach((callback) => { callback(data) });
```